shared/naturalcrit/codeEditor/codeEditor.jsx

```diff
@@ -67,15 +74,40 @@ const CodeEditor = createClass({
 			tabSize           : 2,
 			historyEventDelay : 250,
 			extraKeys         : {
-				'Ctrl-B' : this.makeBold,
-				'Cmd-B'  : this.makeBold,
-				'Ctrl-I' : this.makeItalic,
-				'Cmd-I'  : this.makeItalic,
-				'Ctrl-M' : this.makeSpan,
-				'Cmd-M'  : this.makeSpan,
-				'Ctrl-/' : this.makeComment,
-				'Cmd-/'  : this.makeComment
-			}
+				'Ctrl-B'  : this.makeBold,
+				'Cmd-B'   : this.makeBold,
+				'Ctrl-I'  : this.makeItalic,
+				'Cmd-I'   : this.makeItalic,
+				'Ctrl-M'  : this.makeSpan,
+				'Cmd-M'   : this.makeSpan,
+				'Ctrl-/'  : this.makeComment,
+				'Cmd-/'   : this.makeComment,
+				'Ctrl-\\' : this.toggleCodeFolded,
+				'Cmd-\\'  : this.toggleCodeFolded,
+				'Ctrl-['  : this.foldAllCode,
+				'Cmd-['   : this.foldAllCode,
+				'Ctrl-]'  : this.unfoldAllCode,
+				'Cmd-]'   : this.unfoldAllCode
+			},
+			foldGutter  : true,
+			foldOptions : {
+				rangeFinder : CodeMirror.fold.homebrewery,
+				widget      : (from, to)=>{
+					let text = '';
+					let currentLine = from.line;
+					const maxLength = 50;
+					while (currentLine <= to.line && text.length <= maxLength) {
+						text += this.codeMirror.getLine(currentLine);
+						if(currentLine < to.line) {
+							text += ' ';
+						}
+						currentLine += 1;
+					}
+
+					return `\u21A4${text.substr(0, maxLength)}\u21A6`;
```

Comment:
At this point I'm nitpicking, but:

1) Can we make the text inside another span that is *not* shadowed? It turns into a blurry mess. The little arrows on the side are fine shadowed.
2) Can we `.trim()` the text and add a single space between the arrows and the contents?
3) Not fully decided on this, but perhaps we can add elipses if the length is cut off?

![image](https://user-images.githubusercontent.com/5878534/141338463-a3cc4bcd-0f43-4372-bd96-c5538b8634b6.png)

Reply:
How does this look? Turns out it seemingly sets the innerhtml to the value of the widget function, so we probably can't add additional html elements inside it.

![image](https://user-images.githubusercontent.com/4909307/141380282-99e19493-05c1-45ab-81bf-98f9ef432e06.png)

Reply:
Huh. Ok, yeah this is fine I think.